src/Microsoft.DotNet.Interactive/KernelBase.cs

```diff
@@ -348,10 +348,6 @@ public void RegisterForDisposal(IDisposable disposable)
             _disposables.Add(disposable);
         }
 
-        protected abstract Task HandleSubmitCode(
-            SubmitCode command, 
-            KernelInvocationContext context);
-
         private protected void SetHandler(
```

Comment:
This could be
```
        private protected void SetHandler(
            IKernelCommand command,
            KernelInvocationContext context)
        {
            if (command is KernelCommandBase kb)
            {
                if (kb.Handler == null)
                {
                    switch(command, this)
                    {
                        case (SubmitCode submitCode, IKernelCommandHandler<SubmitCode> submitHandler) :
                            SetHandler(submitHandler, submitCode);
                            break;
                        case (RequestCompletion requestCompletion, IKernelCommandHandler<RequestCompletion> completionHandler):
                            SetCompletionHandler(requestCompletion, completionHandler);
                            break;
                        case (RequestHoverText hoverCommand, IKernelCommandHandler<RequestHoverText> hoverHandler):
                            SetHandler(hoverHandler, hoverCommand);
                            break;
                        case (ChangeWorkingDirectory cwdCommand, _):
                            cwdCommand.Handler = (_, __) =>
                            {
                                Directory.SetCurrentDirectory(cwdCommand.WorkingDirectory.FullName);
                                context.Publish(new WorkingDirectoryChanged(cwdCommand.WorkingDirectory, cwdCommand));
                                return Task.CompletedTask;
                            };
                            break;
                    }
                }
            }
        }
```

Reply:
I'll deflect to @jonsequitur on this one; I have no preference.